### ISSUE TITLE
Allow addition of title and brief documentation in Flat Viewer

### ISSUE THREAD

**Issue by melaniewalsh** [githubocto/flat-viewer#44]
Thank you for Flat Viewer! I love this project.

This is a feature request, and I'm not sure how feasible it would be to implement. But we are using Flat Viewer to host and share datasets for the [Post45 Data Collective](https://data.post45.org/), a repository for literary and cultural data from 1945 to the present. We would love to have  the option of adding a title or perhaps some brief documentation to the top of the Flat Viewer page — perhaps this could be customized by including information in a certain way in a README file in each folder in the repository?

Just to give a concrete example, here's what our current framework for hosting the datasets looks like:
<img width="1433" alt="Screenshot 2023-06-29 at 11 27 27 AM" src="https://github.com/githubocto/flat-viewer/assets/20325102/58397d0e-ea9a-412e-af3d-0b27c91928f5">
vs

<img width="1427" alt="Screenshot 2023-06-29 at 11 28 55 AM" src="https://github.com/githubocto/flat-viewer/assets/20325102/e98d6830-28d0-463f-b562-7bb03b456a60">

We would basically like to add the title, author, and contextual information.

**Comment by RandomFractals** [comment]
@melaniewalsh to refine that feature implementation and UX, what if Flat Viewer checked for an `.md` document with the same filename as your dataset document, loaded and rendered that markdown document with dataset info, data source link, etc. using [markdown-it](https://github.com/markdown-it/markdown-it) library and plugins?

It can default to loading that dataset info from `README.md` in the folder containing your data file or parent folder if the corresponding `<data-file_name>.md` is not found. That would cover cases when a data folder contains multiple data files with different datasets, or when `README.md` is too broad and large to use for the dataset info header to display.

**Comment by melaniewalsh** [comment]
@RandomFractals Yes, that would be fantastic! Thanks so much for helping to refine this idea. I love the idea of using markdown-it and checking for an `.md` file with the same name as the dataset (and defaulting to the `README.md` if not)
